homeassistant/components/shelly/number.py

```diff
@@ -0,0 +1,116 @@
+"""Sensor for Shelly."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Final, cast
+
+import async_timeout
+
+from homeassistant.components.number import NumberEntity, NumberMode
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import AIOSHELLY_DEVICE_TIMEOUT_SEC, CONF_SLEEP_PERIOD, NUMBER_ENTITIES
+from .entity import (
+    BlockAttributeDescription,
+    ShellySleepingBlockAttributeEntity,
+    async_setup_entry_attribute_entities,
+)
+from .utils import get_device_entry_gen
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+SENSORS: Final = {
+    ("device", "valvePos"): BlockAttributeDescription(
+        icon="mdi:pipe-valve",
+        name="Valve Position",
+        unit=PERCENTAGE,
+        available=lambda block: cast(int, block.valveError) != 1,
+        entity_category=EntityCategory.CONFIG,
```

Comment:
All the consts defined in `const.py` should be properties of the entity description, extend the `BlockAttributeDescription` to something like `NumberAttributeDescription` to contain the min/max/step/mode

Reply:
Let's see if the approach I used is what you expected